src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -96,6 +96,10 @@ internal static class RuntimeExceptionHelpers
                         FailFast("Access Violation: Attempted to read or write protected memory. This is often an indication that other memory is corrupt. The application will be terminated since this platform does not support throwing an AccessViolationException.");
                         return null;
 
+                    case ExceptionIDs.IllegalInstruction:
+                        FailFast("Illegal instruction.");
```

Comment:
> Do you think that the intrinsics are the most likely case of illegal instruction in general so that we would like to skew the message towards those?

That should be the only thing causing such cases, outside of JIT or native compiler bugs where they are generating incorrect code bytes themselves.

Reply:
Bad (unsafe) user code can lead to illegal instruction crashes too.

Reply:
👍, I should have said for our own code.

Users manually trying to emit code bytes and execute it, such as via function pointers, can certainly cause it as well.